src/Text/Pandoc/Shared.hs

```diff
@@ -774,6 +777,44 @@ filteredFilesFromArchive zf f =
     fileAndBinary :: Archive -> FilePath -> Maybe (FilePath, BL.ByteString)
     fileAndBinary a fp = findEntryByPath fp a >>= \e -> Just (fp, fromEntry e)
 
+
+--
+-- IANA URIs
+--
+
+-- | Schemes from http://www.iana.org/assignments/uri-schemes.html plus
+-- the unofficial schemes coap, doi, javascript, isbn, pmid
+schemes :: [String]
+schemes = [
+  "aaa", "aaas", "about", "acap", "adiumxtra", "afp", "afs", "aim", "apt",
+  "attachment", "aw", "beshare", "bitcoin", "bolo", "callto", "cap", "chrome",
+  "chrome-extension", "cid", "coap", "com-eventbrite-attendee", "content",
+  "crid", "cvs", "data", "dav", "dict", "dlna-playcontainer", "dlna-playsingle",
+  "dns", "doi", "dtn", "dvb", "ed2k", "facetime", "feed", "file", "finger",
+  "fish", "ftp", "geo", "gg", "git", "gizmoproject", "go", "gopher", "gtalk",
+  "h323", "hcp", "http", "https", "iax", "icap", "icon", "im", "imap", "info",
+  "ipn", "ipp", "irc", "irc6", "ircs", "iris", "iris.beep", "iris.lwz",
+  "iris.xpc", "iris.xpcs", "isbn", "itms", "jar", "javascript", "jms", "keyparc",
+  "lastfm", "ldap", "ldaps", "magnet", "mailto", "maps", "market", "message",
+  "mid", "mms", "ms-help", "msnim", "msrp", "msrps", "mtqp", "mumble", "mupdate",
+  "mvn", "news", "nfs", "ni", "nih", "nntp", "notes", "oid", "opaquelocktoken",
+  "palm", "paparazzi", "platform", "pmid", "pop", "pres", "proxy", "psyc",
+  "query", "res", "resource", "rmi", "rsync", "rtmp", "rtsp", "secondlife",
+  "service", "session", "sftp", "sgn", "shttp", "sieve", "sip", "sips", "skype",
+  "smb", "sms", "snmp", "soap.beep", "soap.beeps", "soldat", "spotify", "ssh",
+  "steam", "svn", "tag", "teamspeak", "tel", "telnet", "tftp", "things",
+  "thismessage", "tip", "tn3270", "tv", "udp", "unreal", "urn", "ut2004",
+  "vemmi", "ventrilo", "view-source", "webcal", "ws", "wss", "wtai", "wyciwyg",
+  "xcon", "xcon-userid", "xfire", "xmlrpc.beep", "xmlrpc.beeps", "xmpp", "xri",
+  "ymsgr", "z39.50r", "z39.50s"
+  ]
+
+-- | Check if the string is a valid URL with a IANA or frequently used but
+-- unofficial scheme (see @schemes@).
+isIanaURI :: String -> Bool
```

Comment:
Why don't we just call this `isURI`?  `IANA` is misleading anyway because of the unofficial ones.  And if we call it `isURI` this will encourage us to use it throughout the code base, which I assume we'd want?